src/Spectre.libStatistics.Tests/StatisticsTest.cpp

```diff
@@ -53,6 +53,16 @@ TEST(MeanTest, mean_of_empty_is_zero)
     EXPECT_THAT(Mean(empty), 0.);
 }
 
+TEST(MedianTest, calculates_vector_median)
+{
+    EXPECT_THAT(Median(data), 2.);
+}
+
+TEST(MedianTest, throws_on_empty_vector)
+{
+    EXPECT_THROW(MedianAbsoluteDeviation(empty), spectre::core::exception::EmptyArgumentException);
+}
```

Comment:
This should be in line with other functions, esp. `Mean(empty)` above, and return 0 imo.

How do you think @Daroslaw @gmrukwa?

Reply:
According to some random experts on statistics:
"[If you have an empty set, then it will have neither a mean, nor a median, nor a mode.](https://www.algebra.com/algebra/homework/Probability-and-statistics/Probability-and-statistics.faq.question.499197.html)"
So I personally see two solutions here; in both functions either throw exception or return `NaN`, with bias on the second solution as it is exactly how Matlab behaves, so I suppose if a software has "Mat" in its name then it has some authority to form opinions about mathematics.

Reply:
I'd go for consistency. If we want to go for `Median(empty)` throwing an exception, we should throw exception in each of the functions. Otherwise we will wonder each time, what's the expected behavior. Follow rule of least astonishment :)

On the other hand I know, empty set has no such statistics, however I had simplicity of use in my mind. Otherwise we'll have to do a lot of checks almost at each call point, just to check whether the set is nonempty. I followed DRY, knowing the possible applications.

Reply:
Do we vote now or what?

Reply:
Discussed during standup meeting; median should return 0 for empty vector, and this behavior should be documented in code.